### PR TITLE
#167313424 Build out user track comments edit history

### PR DESCRIPTION
**Pull request by nshutijonathan**
#### What does this PR do?
-Has tracking comments edit functionality
#### Description of Task to be completed?
- Added functionality for the user to track comments edit history on comment updation
- Tested the comments route
This PR has the following endpoints working 
- UPDATE/http://localhost:8000/api/v1/comments/</id>
#### How should this be manually tested?
- Download or clone the tutorial project code from this repo, in your terminal then git checkout `ft-comments-track-previous-edit-167313424`, install all required NPM packages by running NPM install from the command line in the project root folder (where the package.json is located).
- Start the API by running NPM start from the command line in the project root folder, you should see the message Server listening on port 8000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
- to test authentication and authorization, add this to the header
Key: :x-access-token,value:JWT TOKEN
#### Any background context you want to provide?
 - Implemented the endpoint logic using Sequelize framework 
#### What are the relevant pivotal tracker stories?
[#167313424](https://www.pivotaltracker.com/n/projects/2372121/stories/167313424)
#### Screenshots (if appropriate)

This photo below shows how I get edit history  over the comment, the field called commentRevisions shows the previous comment edited

image

image

image

Below is the swagger documentation screenshot

image

### CODE REVIEW

File: test/comments.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  it('should  comment on article', (done) => {
+    chai
+      .request(server)
+      .post('/api/v1/comments/fakeslug')
+      .send({
+        body: 'I commented',
+      })
+      .set('x-access-token', adminToken)
+      .end((error, res) => {
+        console.log('######', res.body);
```
Comment (Anguandia) [inline test/comments.test.js]: please remove the console.logs

**Review by Anguandia — DISMISSED** [review]

File: src/config/config.js [hunk 2]
```diff
@@ -26,7 +26,8 @@ module.exports = {
     database: DB_NAME_TEST,
     host: DB_HOST,
     dialect: 'postgres',
-    port: DB_PORT
+    port: DB_PORT,
+    logging: false
```
Comment (habinezadalvan) [inline src/config/config.js]: You forgot to remove logging: false line. @Mnickii said that logs should be there for the debugging purpose. Please remove it.

Comment (nshutijonathan) [reply src/config/config.js]: @habinezadalvan I have removed it thank You for reminding me

File: test/comments.test.js (hunk 1 above)
Comment (nshutijonathan) [reply test/comments.test.js]: @Anguandia I have removed the console.logs , Thank You for the feedback

File: src/config/config.js (hunk 2 above)
Comment (habinezadalvan) [reply src/config/config.js]: great

**Review by habinezadalvan — DISMISSED** [review]

File: test/comments.test.js (hunk 1 above)
Comment (Anguandia) [reply test/comments.test.js]: Thanks

**Review by Anguandia — DISMISSED** [review]

**Review by salviosage — DISMISSED** [review]
LGTM

**Review by minega25 — DISMISSED** [review]
LGTM

File: src/controllers/comments.controller.js [hunk 3]
```diff
@@ -134,23 +124,21 @@ class Comments {
...
     const { body } = req.body;
-    const updateComment = await commentsService.updateComment(req.params.id, { body });
-    return res.status(200).send({
-      status: 200,
-      message: 'Updation is successfully',
-      updateComment
-    });
+    const commentRevisions = getComment.dataValues.body;
+    const updateComment = await commentsService.updateComment(req.params.id, { body, commentRevisions });
+    await util.setSuccess(200, 'Updation is successfully', updateComment);
```
Comment (Mnickii) [inline src/controllers/comments.controller.js]: Update is successful.

Comment (nshutijonathan) [reply src/controllers/comments.controller.js]: @I have returned update is successfully instead of Updation is successful as You suggested, Thank You

**Review by nkalyesubula — DISMISSED** [review]

**Pull request merged**